### ISSUE TITLE
After enabling appsignal, ActiveJob queue changed to 'default'

### ISSUE THREAD

**Issue by JasonBarnabe** [appsignal/appsignal-ruby#827]
In an initializer, we have:

```
Rails.application.config.active_job.default_queue_name = :medium
```

After adding AppSignal, any job that didn't specify a queue using `queue_as` in the class started going to the "default" queue rather than "medium". That sucked because we didn't have any sidekiq workers looking at that queue.

Ruby 2.7.5
Rails 6.1.4.6
AppSignal 3.0.21

**Comment by tombruijn** [comment]
Hi @JasonBarnabe,

I've reproducing the situation you're describing, but no luck unfortunately. I've updated our Rails 6 + Sidekiq test app and configured it as you described. The jobs without an explicit `queue_as` (specified as the `default_queue_name`) are being picked up by Sidekiq. I don't see them being queued in the "default" queue.

The only way I can trigger this behavior so far is by queuing a Sidekiq job without ActiveJob, that defaults to the "default" queue. Are the jobs you're seeing in the "default" queue ActiveJob jobs or purely Sidekiq jobs?

Changes I made to the test app: https://github.com/appsignal/test-setups/compare/ruby-827-activejob-sidekiq-default-queue-name

**Comment by tombruijn** [comment]
Yes now I can reproduce it. That's really weird. We don't track the ActiveJob creation or interfere with any Rails config, so I'm not sure how loading the AppSignal gem breaks this.

For now you can use your workaround or move it to the `config/application.rb` file. What is the reason you moved this config to the initializer?

**Comment by tombruijn** [comment]
Let's look at a quick fix first. It may not be perfect, but this is a serious bug that we need to fix now and make our code "nice" again later.

**Comment by JasonBarnabe** [comment]
> For now you can use your workaround or move it to the config/application.rb file. What is the reason you moved this config to the initializer?

We already had an initializer for ActiveJob monkeypatches - seemed to make sense to keep it all ActiveJob config together.

**Comment by tombruijn** [comment]
Hi @JasonBarnabe, we've published a fix for this in version 3.0.23 of the Ruby gem. Can you upgrade and let us know if it fixes the issue for you, if you remove the workaround?
